src/libsystemd/sd-device/device-enumerator.c

```diff
@@ -995,6 +996,66 @@ _public_ sd_device *sd_device_enumerator_get_subsystem_next(sd_device_enumerator
         return enumerator->devices[++enumerator->current_device_index];
 }
 
+int device_enumerator_scan_devices_and_subsystems(sd_device_enumerator *enumerator) {
+        int r = 0, k;
+
+        assert(enumerator);
+
+        if (enumerator->scan_uptodate &&
+            enumerator->type == DEVICE_ENUMERATION_TYPE_ALL)
+                return 0;
+
+        device_enumerator_unref_devices(enumerator);
+
+        if (!set_isempty(enumerator->match_tag)) {
+                k = enumerator_scan_devices_tags(enumerator);
+                if (k < 0)
+                        r = k;
+        } else if (enumerator->match_parent) {
+                k = enumerator_scan_devices_children(enumerator);
+                if (k < 0)
+                        r = k;
+        } else {
+                const char *subsysdir;
+
+                if (access("/sys/subsystem", F_OK) >= 0)
```

Comment:
i wonder if we should drop this stuff. `/sys/subsystem` is preparation for a future that never came. And given that the main proponent of this left Linux kernel development (Kay), I doubt this will ever come. So maybe we should start dropping references to /sys/subsystem/ given it's unlikely to materialize anytime soon.

Reply:
(but this is just a general comment, not really an issue with this PR)